zlib-rs/src/inflate.rs

```diff
@@ -280,9 +280,10 @@ struct Table {
     bits: usize,
 }
 
-pub(crate) struct State<'a> {
+pub struct State<'a> {
     /// Current inflate mode
-    mode: Mode,
+    pub mode: Mode,
+
```

Comment:
hmm yeah we really don't want to make this public. The options to work around that are kind of limited though. I think it's best to make some `pub unsafe fn set_mode_dict(strm: &mut z_stream)` to inflate. We can hide that funtion in the future with feature flags.